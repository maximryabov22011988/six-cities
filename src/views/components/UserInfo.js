import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { BASE_URL } from '../../api';

const propTypes = {
  isAuth: PropTypes.bool.isRequired,
  avatarUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  email: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, avatarUrl, email } = this.props;

    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <Link
            to="/favorites"
            className="header__nav-link header__nav-link--profile"
          >
            <div className="header__avatar-wrapper user__avatar-wrapper">
              <img
                className="user__avatar"
                alt="avatar"
                src={`${BASE_URL}${avatarUrl}`}
              />
            </div>
            <span className="header__user-name user__name">{email}</span>
          </Link>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
