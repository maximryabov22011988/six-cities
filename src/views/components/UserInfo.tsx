import * as React from 'react';

import { Link } from 'react-router-dom';

import Avatar from './Avatar';

import { BASE_URL } from '../../api';

const propTypes = {
  avatarUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  email: PropTypes.oneOfType([PropTypes.string, PropTypes.any]),
  isAuth: PropTypes.bool.isRequired,
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, avatarUrl, email } = this.props;

    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <Link className="header__nav-link header__nav-link--profile" to="/favorites">
            <Avatar alt="User avatar" name={email} parentClassName="header" src={`${BASE_URL}${avatarUrl}`} />
          </Link>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
