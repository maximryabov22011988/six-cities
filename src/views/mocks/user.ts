export default {
  avatar_url: '/static/avatar/8.jpg',
  email: 'Oliver.conner@gmail.com',
  id: 1,
  is_pro: false,
  name: 'Oliver.conner',
};
