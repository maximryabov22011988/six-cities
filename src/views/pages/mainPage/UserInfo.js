import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const propTypes = {
  isAuth: PropTypes.bool.isRequired,
  email: PropTypes.string.isRequired
};

class UserInfo extends React.PureComponent {
  render() {
    const { isAuth, email } = this.props;
    return (
      <li className="header__nav-item user">
        {!isAuth ? (
          <Link to="/login">Sign In</Link>
        ) : (
          <a className="header__nav-link header__nav-link--profile" href="#">
            <div className="header__avatar-wrapper user__avatar-wrapper" />
            <span className="header__user-name user__name">{email}</span>
          </a>
        )}
      </li>
    );
  }
}

UserInfo.propTypes = propTypes;

export default UserInfo;
