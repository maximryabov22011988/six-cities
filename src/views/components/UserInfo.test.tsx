import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import UserInfo from './UserInfo';
import user from '../mocks/user';

it('UserInfo renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <UserInfo avatarUrl={user.avatar_url} email={user.email} isAuth={true} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
