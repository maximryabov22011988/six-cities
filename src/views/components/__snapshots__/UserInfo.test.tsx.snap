// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo renders correctly with mandatory props 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="/favorites"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper false"
    >
      <img
        alt="User avatar"
        className="reviews__avatar user__avatar"
        src="https://es31-server.appspot.com/six-cities/static/avatar/8.jpg"
      />
    </div>
    <span
      className="header__user-name"
    >
      Oliver.conner@gmail.com
    </span>
  </a>
</li>
`;
